File: easyweb/src/main/java/com/lzx/easyweb/cache/interceptor/Chain.java
package com.lzx.easyweb.cache.interceptor;

import java.util.List;


public class Chain {

    private List<ResourceInterceptor> mInterceptors;
    private int mIndex = -1;
    private CacheRequest mRequest;

    Chain(List<ResourceInterceptor> interceptors) {
        mInterceptors = interceptors;
    }

    public WebResource process(CacheRequest request) {
        if (++mIndex >= mInterceptors.size()) {
            return null;
        }
        mRequest = request;
        ResourceInterceptor interceptor = mInterceptors.get(mIndex);
        return interceptor.load(this);
    }

    public CacheRequest getRequest() {
        return mRequest;
    }
}

File: easyweb/src/main/java/com/lzx/easyweb/cache/interceptor/Chain.kt
package com.lzx.easyweb.cache.interceptor

import com.lzx.easyweb.cache.CacheRequest

class Chain internal constructor(private val mInterceptors: MutableList<ResourceInterceptor?>) {
    private var mIndex = -1
    var request: CacheRequest? = null
        private set

    fun process(request: CacheRequest?): WebResource? {
        if (++mIndex >= mInterceptors.size) {
            return null
        }
        this.request = request
        val interceptor = mInterceptors[mIndex]
        return interceptor?.load(this)
    }

}